remoteappmanager/handlers/base_handler.py

```diff
@@ -40,3 +41,20 @@ def render(self, template_name, **kwargs):
 
         args.update(kwargs)
         super(BaseHandler, self).render(template_name, **args)
+
+    def write_error(self, status_code, **kwargs):
+        """Render error page for uncaught errors"""
+        status_message = responses.get(status_code, 'Unknown HTTP Error')
+        message = ""
+
+        # If this error was caused by an uncaught exception
+        # log exception message and reference number as well
+        exc_info = kwargs.get('exc_info')
+        if exc_info:
+            exception = exc_info[1]
+            ref = self.log.issue(status_message, exception)
+            reason = getattr(exception, 'reason', '')
+            message = '{} Ref.: {}'.format(reason, ref)
```

Comment:
> Tornado, on the other hand, seems to favour the reason usage only if you use non-standard codes.
In their documentation, by 'non-standard codes', they mean 'non-standard status code' and therefore no standard reason phrase can be found (say a status code 420).  In many places in their own code, they custom the 'reason' for common status codes such as 403. (See `StaticFilePathHandler. validate_absolute_path`)